.github/workflows/build.yml

```diff
@@ -81,33 +54,69 @@ jobs:
           GITHUB_REF_NAME_NORMALIZED=$(echo ${{github.ref_name}} | tr / -)
           echo "GITHUB_REF_NAME_NORMALIZED=${GITHUB_REF_NAME_NORMALIZED}" >> $GITHUB_ENV
 
+          DOCKERIZE=${{ contains(fromJSON('["master", "staging", "production"]'), github.ref_name) || startsWith(github.event.head_commit.message, 'dockerize')}}
+          echo "DOCKERIZE=${DOCKERIZE}" >> $GITHUB_ENV
       - name: Setup Node
         uses: actions/setup-node@v3
         with:
-          node-version: "${{ env.NODE_VERSION }}"
-          cache: "yarn"
+          node-version-file: package.json
+          # Caching yarn dir & running yarn install is too slow
+          # Instead, we aggressively cache node_modules below to avoid calling install
 
       - name: Get cached node modules
         id: cache
         uses: actions/cache@v3
         with:
           path: |
             **/node_modules
-          key: ${{ env.DEPENDENCIES_CACHE }}-${{ hashFiles('yarn.lock') }}
+          key: node_modules-${{ runner.arch }}-${{ env.NODE_VERSION }}-${{ hashFiles('yarn.lock') }}
 
-      - name: Yarn install
+      - name: Install node_modules
         if: steps.cache.outputs.cache-hit != 'true'
-        run: yarn install --frozen-lockfile --immutable && yarn cache clean
+        run: yarn install --immutable
 
-      - name: Build the application
+      - name: Build the DBs
         run: |
-          cp ${{ env.PARABOL_BUILD_ENV_PATH }} ./.env && \
-          yarn db:migrate && \
-          yarn pg:migrate up && \
-          yarn pg:build && \
-          yarn build --no-deps
+          cp ${{ env.PARABOL_BUILD_ENV_PATH }} ./.env
+          yarn db:migrate
+          yarn pg:migrate up
+          yarn pg:build
+
+      - name: Build cheap
+        if: env.DOCKERIZE != 'true'
+        run: yarn build
+      - name: Build prod
+        if: env.DOCKERIZE == 'true'
+        run: yarn build --no-deps
+
+      - name: Check Code Quality
+        run: yarn codecheck
```

Comment:
our packages are so tightly coupled that nx affected would show all 3 of them being invalided.
nx affected required a git fetch-depth of 0.
instead, we can run all 3 code checks in parallel on every push & it'll be faster than before